src/enigma2/data/Channel.cpp

```diff
@@ -222,7 +222,11 @@ bool Channel::HasRadioServiceType()
   if (found != std::string::npos)
     radioServiceType = radioServiceType.substr(0, found);
 
-  return radioServiceType == RADIO_SERVICE_TYPE;
+  if (radioServiceType.empty())
+    return false;
+
+  char e2ServiceType = toupper(radioServiceType[0]);
```

Comment:
Why call toupper? Have you ever see an instance where the values are given in lower case?

The implementation could be simpler by using string instead of char. And if lower case is possible simply adding a third item to the list. But I’m still not convinced lowercase `A` can happen.

Reply:
The case should always be lower case. if the service reference is untouched and coming straigth from within enigma2.
Their encode function uses `snprintf(hex, 8, "%02x", c);` but this never seems to make it to us, bcs its also stripping the leading 0 which would normally be prepended here in the case of 0xa.
https://github.com/openatv/enigma2/blob/7.0/lib/service/service.cpp#L20

So yeah not sure whats altering the service reference after this but enigma2 itself supports reading lower and upper case hex strings due to the use of the '%x' formatting character.
https://github.com/openatv/enigma2/blob/7.0/lib/service/service.cpp#L36
Thats why i opted to make this case insensitive, if enigma eats it, we should too.

So yeah unsolved mystery, not sure whos touching things before we end up reading them and thats why i went for the easy route. If you can say for certain that alpha chars are always lowercase we can strip this part.

---
True the 0x02 for radio (ends up being :2: in our case) is never going to clash with the 0x1X prefixes anyway so i could have just compared the entire string. Gonna fix that later!

Semantically `tolower` would probably be better, imma adjust this and use std::array as container.
You have the final verdict on the to lower case or not to lower case.

Reply:
You could just have the string in the array be: “2”, “A” and “a” if you would rather be on the safe side.

Reply:
Yeah sure, adding both of them then, i just wasnt sure before.